lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -214,43 +215,185 @@ bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
+                TF_WARN(
+                    "The file: %s could not be found within the USDZ archive for extraction.",
+                    filePath.c_str());
+                return false;
+            }
+            const char*          fileData = it.GetFile();
+            UsdZipFile::FileInfo fileInfo = it.GetFileInfo();
+
+            bool             needsUniqueFilename = false;
+            std::string_view sFile(fileData, fileInfo.size);
+            char             md5Digest[32] = { 0 };
+            UsdMayaHashUtil::GenerateMD5DigestFromByteStream(
+                reinterpret_cast<const uint8_t*>(fileData), sFile.size(), md5Digest);
+            std::unordered_map<std::string, std::string>::iterator itExistingHash
+                = UsdMayaReadUtil::mapFileHashes.find(unresolvedFilePath);
+            if (itExistingHash
+                == UsdMayaReadUtil::mapFileHashes.end()) { // NOTE: (yliangsiew) Means that the
+                                                           // texture hasn't been extracted before.
+                UsdMayaReadUtil::mapFileHashes.insert(
+                    { unresolvedFilePath,
+                      std::string(
+                          md5Digest) }); // NOTE: (yliangsiew) This _should_ be the common case.
+            } else {
+                std::string existingHash = itExistingHash->second;
+                if (memcmp(md5Digest, existingHash.c_str(), 32 * sizeof(char)) == 0) {
+                    TF_WARN(
+                        "A duplicate texture: %s was found, skipping extraction of it and re-using "
+                        "the existing one.",
+                        unresolvedFilePath.c_str());
+                    unresolvedFilePath = itExistingHash->first;
+                } else {
+                    // NOTE: (yliangsiew) Means that a duplicate texture with the same name but with
+                    // different contents was found. Instead of failing, continue extraction with a
+                    // different filename instead and point to that one.
+                    needsUniqueFilename = true;
+                }
+            }
+
+            // NOTE: (yliangsiew) Write the file to disk now.
+            char filename[FILENAME_MAX] = { 0 };
+            memcpy(filename, unresolvedFilePath.c_str(), unresolvedFilePath.size());
+            memset(filename + unresolvedFilePath.size(), 0, 1);
+            UsdMayaUtilFileSystem::pathStripPath(filename);
+
+            char extractedFilePath[FILENAME_MAX] = { 0 };
+            memcpy(
+                extractedFilePath,
+                jobArgs.importUSDZTexturesFilePath.c_str(),
+                jobArgs.importUSDZTexturesFilePath.size());
+            memset(extractedFilePath + jobArgs.importUSDZTexturesFilePath.length(), 0, 1);
+            bool bStat = UsdMayaUtilFileSystem::pathAppendPath(extractedFilePath, filename);
+            TF_VERIFY(bStat);
+
+            if (needsUniqueFilename) {
+                int counter = 0;
```

Comment:
@ysiewappl Not really a review because I was just checking the new logic - I'm assuming you intended to increment this `counter` somewhere?